backend-lib/src/main/rust/lib.rs

```diff
@@ -2071,16 +2143,28 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_addProofs
         let pczt = Pczt::parse(&env.convert_byte_array(pczt)?[..])
             .map_err(|e| anyhow!("Invalid PCZT: {:?}", e))?;
 
-        let spend_params = utils::java_string_to_rust(env, &spend_params);
-        let output_params = utils::java_string_to_rust(env, &output_params);
-        let prover = LocalTxProver::new(Path::new(&spend_params), Path::new(&output_params));
+        let mut prover = Prover::new(pczt);
 
-        let pczt_with_proofs = Prover::new(pczt)
-            .create_orchard_proof(&orchard::circuit::ProvingKey::build())
-            .map_err(|e| anyhow!("Failed to create Orchard proof for PCZT: {:?}", e))?
-            .create_sapling_proofs(&prover, &prover)
-            .map_err(|e| anyhow!("Failed to create Sapling proofs for PCZT: {:?}", e))?
-            .finish();
+        if prover.requires_orchard_proof() {
+            prover = prover
+                .create_orchard_proof(&orchard::circuit::ProvingKey::build())
+                .map_err(|e| anyhow!("Failed to create Orchard proof for PCZT: {:?}", e))?;
+        }
+        assert!(!prover.requires_orchard_proof());
```

Comment:
In a multisig context, is it always the case that a single entity will be responsible for adding all the proofs?

Reply:
If it isn't, that would be a separate enhancement.